library/common-library/templates/_hostnetwork.tpl

```diff
@@ -7,20 +7,41 @@ Returns "true" if `hostNetwork` is enabled, otherwise fallbacks to a overridable
 {{- /* This allows us to use `$global` as an empty dict directly in case `Values.global` does not exists */ -}}
 {{- $global := index .Values "global" | default dict -}}
 
-{{- /* `get` will return "" (empty string) if value is not found, and the value otherwise, so we can type-assert with kindIs */ -}}
+{{- /*
+`get` will return "" (empty string) if value is not found, and the value otherwise, so we can type-assert with kindIs
+
+We also want only to return when this is true, returning `false` here will template "false" (string) when doing
+an `(include "common.hostNetwork" .)`, which is not an "empty string" so it is `true` if it is used
+as an evaluation somewhere else.
+*/ -}}
 {{- if get .Values "hostNetwork" | kindIs "bool" -}}
-    {{- .Values.hostNetwork -}}
+    {{- if .Values.hostNetwork -}}
+        {{- .Values.hostNetwork -}}
+    {{- end -}}
 {{- else if get $global "hostNetwork" | kindIs "bool" -}}
-    {{- $global.hostNetwork -}}
+    {{- if $global.hostNetwork -}}
+        {{- $global.hostNetwork -}}
+    {{- end -}}
 {{- else -}}
     {{- include "common.hostNetwork.defaultOverride" . -}}
 {{- end -}}
 {{- end -}}
 
 
+{{- /*
+Abstraction of the hostNetwork toggle.
```

Comment:
I do not understand why this is useful.. To get an explicit false instead of ""?

Reply:
Because if you use something like `{{ if include "common.hostNetwork" . }}` and it returns `"false"` (note it is a string) helm evaluates it as `true`. Because Helm falsiness does not include `"false"` but an empty string.

Reply:
Should I add [this comment](https://github.com/newrelic/helm-charts/blob/fix/common-library/hostnetwork/library/common-library/templates/_low-data-mode.tpl#L10-L14) that is in `lowDataMode` and `staging` also here?

Reply:
it it fine!